src/dft/backends/mklcpu/commit.cpp

```diff
@@ -28,22 +28,97 @@
 
 #include "oneapi/mkl/dft/detail/mklcpu/onemkl_dft_mklcpu.hpp"
 
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "mkl_service.h"
+#include "mkl_dfti.h"
+
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace mklcpu {
 
-void commit_f(descriptor<precision::SINGLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
-}
-void commit_c(descriptor<precision::SINGLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
-}
-void commit_d(descriptor<precision::DOUBLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
-}
-void commit_z(descriptor<precision::DOUBLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
+class commit_derived_impl : public oneapi::mkl::dft::detail::commit_impl {
+public:
+    commit_derived_impl(sycl::queue queue, dft_values config_values)
+            : oneapi::mkl::dft::detail::commit_impl(queue),
+              status(0) {
+        logf("CPU impl, handle->%p", &handle);
+
+        DFTI_DESCRIPTOR_HANDLE local_handle = nullptr;
+
+        std::cout << config_values << std::endl;
+        if (config_values.rank == 1) {
+            status = DftiCreateDescriptor(&local_handle, precision_map[config_values.precision],
+                                          domain_map[config_values.domain], config_values.rank,
+                                          config_values.dimension[0]);
+        }
+        else {
+            status = DftiCreateDescriptor(&local_handle, precision_map[config_values.precision],
+                                          domain_map[config_values.domain], config_values.rank,
+                                          &config_values.dimension[0]);
+        }
+        if(status != DFTI_NO_ERROR) throw oneapi::mkl::exception("dft", "commit", "DftiCreateDescriptor failed");
+
+        set_value(local_handle, config_values);
+
+        status = DftiCommitDescriptor(local_handle);
+        if(status != DFTI_NO_ERROR) throw oneapi::mkl::exception("dft", "commit", "DftiCommitDescriptor failed");
+
+        // commit_impl (pimpl_->handle) should return this handle
+        handle = local_handle;
+    }
+
+    commit_derived_impl(const commit_derived_impl* other)
+            : oneapi::mkl::dft::detail::commit_impl(*other) { }
+
+    virtual oneapi::mkl::dft::detail::commit_impl* copy_state() override {
+        return new commit_derived_impl(this);
+    }
```

Comment:
Is this safe? They look like they share a descriptor handle, which might to DtfiFreeDescriptor being called twice on the same handle.

Reply:
Where is this even needed?

Reply:
deferred for later